### DISCUSSION TITLE
Setting Laravel Blade variables from `Inertia::render`

### DISCUSSION THREAD

**Discussion by mattgreenfield** [inertiajs/inertia#1348] [Help] [answered]
I have a layout file that I'm using for my existing Laravel app and now I want to add Vue to one page of the app.

I can pass this existing blade layout file to Inertia but it's complaining that the variables are undefined (e.g. `$title`).

Is there a way to render Inertia in the middle of a page rendered with Blade? The motivation is that I don't want to rebuild my header and footer in vue or to have to maintain two versions.

Thanks

**Comment by giorgiopogliani** [comment, marked as answer]
Did you pass the title variable? I didn't tested it right now but I think it should work. 

For the header and footer you could extract them in a blade component with its class. In the classes you could make all the queries you need and pass the props.

**Reply by mattgreenfield** [reply]
Thanks for the reply. Yes, I've tried setting the variable but it doesn't work, I assume it's only setting it in JS and not in Blade.
```php
return Inertia::render('MyPage', [
            'title' => 'This is my page',
        ]);
```

Good point about the header and footer components, I might have a lot of refactoring to do!

**Reply by giorgiopogliani** [reply]
Tried now and it's all wrapped in a `$page` variable so to access the title in blade you could do this!
```blade 
{{ $title ?? $page['props']['title'] ?? config('app.name', 'Laravel') }}
```

**Reply by mattgreenfield** [reply]
Ah that's great! thank you

**Reply by bcheidemann** [reply]
It would be nice if there was a way to do this without duplicating the content in the `data-page` prop. My use case is that I want to add open graph meta tags to certain pages which are (other than the shell) client rendered. I don't really want to add the meta tags both in the header and in the `data-page` prop.
